language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/L.hs

```diff
@@ -294,11 +294,10 @@ internalEvaluateL t = computeL [] emptyHeap (Closure t mempty)
 translateResult :: Functor f => f LMachineResult -> f (Plain Term)
 translateResult = fmap $ \(Closure t _, _) -> t
 
--- | Evaluate a term using the L machine. May throw an 'LMachineException'.
-evaluateL :: Term TyName Name () -> EvaluationResultDef
-evaluateL = either throw translateResult . extractEvaluationResult . internalEvaluateL
+-- | Evaluate a term using the L machine.
+evaluateL :: Term TyName Name () -> Either LEvaluationException (Term TyName Name ())
```

Comment:
Exposed the same interface in all the machines. `evaluate*` returns an `Either *EvaluationException (Term TyName Name ())`, `unsafeEvaluate*` returns an `EvaluationResultDef`.